### PR TITLE
cli: add find_arxiv_duplicates command INSPIR-1835

### PR DESCRIPTION
**Pull request by ammirate**
Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

### CODE REVIEW

File: inspirehep/modules/records/cli.py [hunk 1]
```diff
@@ -354,6 +356,67 @@ def handle_duplicates(remove_no_control_number, remove_duplicates,
...
+    workflows_ids = [
+        wf_id[0] for wf_id in WorkflowObjectModel.query.
+            with_entities(WorkflowObjectModel.id).
+            filter(WorkflowObjectModel.created > starting_date).
+            filter(WorkflowObjectModel.status == ObjectStatus.COMPLETED).
+            all()
+    ]
+    duplicates = dict()
+
+    with click.progressbar(length=len(workflows_ids), label='Processing workflows') as bar:
```
Comment (michamos) [inline inspirehep/modules/records/cli.py]: that's a strange way to do it. Why not simply
```suggestion
    with click.progressbar(workflow_ids, label='Processing workflows') as bar:
```
followed by
```
    for wf_id in bar:
```
then you don't have to update the bar by hand.

File: inspirehep/modules/records/cli.py [hunk 2]
```diff
@@ -354,6 +356,67 @@ def handle_duplicates(remove_no_control_number, remove_duplicates,
...
+        for wf_id in workflows_ids:
+            wf = workflow_object_class.get(wf_id)
+            recid = wf.data.get('control_number')
+
+            if not recid:
```
Comment (michamos) [inline inspirehep/modules/records/cli.py]: you should be able to do this filtering in the DB by adding something like `.filter(WorkflowObjectModel.data).has_key('control_number')` (might require a `type_coerce` to `JSON` to work). That should be way faster.

File: inspirehep/modules/records/cli.py [hunk 3]
```diff
@@ -354,6 +356,67 @@ def handle_duplicates(remove_no_control_number, remove_duplicates,
...
+                bar.update(1)
+                continue  # rejected paper
+
+            if wf.data.get('arxiv_eprints'):
```
Comment (michamos) [inline inspirehep/modules/records/cli.py]: you could add a similar test for the presence of `arxiv_eprints` to the query.

File: inspirehep/modules/records/cli.py [hunk 4]
```diff
@@ -354,6 +360,72 @@ def handle_duplicates(remove_no_control_number, remove_duplicates,
     db.session.commit()
 
 
+@check.command()
+@with_appcontext
+def find_arxiv_duplicates(starting_date=None):
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply inspirehep/modules/records/cli.py]: IMHO it's not necessary to call this CLI command with parameter at all. Basically it should swipe whole db. Specifying a date here is very specific use-case which shouldn't be available from CLI. It still can be used when calling this as a function from inspire shell.

**Pull request merged**